osconfig_tests/gcp_clients/gcp_clients.go

```diff
@@ -28,37 +29,57 @@ import (
 
 var (
 	storageClient  *storage.Client
+	computeClient  compute.Client
 	osconfigClient *osconfig.Client
 
 	populateClientOnce sync.Once
 )
 
-// PopulateClients creates singleton clients for various services
-func PopulateClients(ctx context.Context) {
+func populateClients(ctx context.Context) error {
+	var err error
 	populateClientOnce.Do(func() {
-		createOsConfigClient(ctx)
-		createStorageClient(ctx)
+		err = createComputeClient(ctx)
+		err = createOsConfigClient(ctx)
+		err = createStorageClient(ctx)
 	})
+	return err
 }
 
-func createStorageClient(ctx context.Context) {
+func createComputeClient(ctx context.Context) error {
+	var err error
+	computeClient, err = compute.NewClient(ctx, option.WithCredentialsFile(config.OauthPath()))
+	if err != nil {
+		return err
+	}
+	return nil
```

Comment:
You could remove the previous 4 lines and just `return err` after creating the client. Here and a few places below.